common/background_exception_catcher.h

```diff
@@ -0,0 +1,117 @@
+#pragma once
+
+// BoB robotics includes
+#include "assert.h"
+
+// Standard C includes
+#include <csignal>
+
+// Standard C++ includes
+#include <array>
+#include <stdexcept>
+#include <string>
+#include <unordered_set>
+
+namespace BoBRobotics {
+//----------------------------------------------------------------------------
+// BoBRobotics::BackgroundExceptionCatcher
+//----------------------------------------------------------------------------
+//! A wrapper for passing exceptions between threads (i.e. a background thread to the main one)
+class BackgroundExceptionCatcher
+{
+public:
+    BackgroundExceptionCatcher()
+    {
+        // We can have only one instance
+        BOB_ASSERT(!CatcherExists);
+    }
+
+    ~BackgroundExceptionCatcher()
+    {
+        // Unregister signal handlers
+        for (auto sig : m_Signals) {
+            std::signal(sig, SIG_DFL);
+        }
+
+        CatcherExists = false;
+    }
+
+#ifdef DEBUG
```

Comment:
Having function signatures in #ifdefs never seems the best plan...How about?
```suggestion
#ifdef DEBUG
#define DEFAULT_TRAPPED_SIGNALS { SIGINT }
#else
#define DEFAULT_TRAPPED_SIGNALS { SIGSEGV, SIGINT, SIGFPE }
#endif
void trapSignals(const std::unordered_set<int> &signals = DEFAULT_TRAPPED_SIGNALS)
{
}
```

Reply:
This is def less gross than my way :-)